doc/release-notes/10981-published-datasets-should-contain-files.md

```diff
@@ -0,0 +1,11 @@
+## Feature: Prevent publishing Datasets without files
```

Comment:
The optional false by default has been fully tested and does work. I think if we add a feature flag then someone can shut off all dataverse collections from requiring files which could cause confusion to the creator of a dataverse expecting files to be required.

Reply:
I meant a flag that has to be set before you can toggle the requirement on/off by collection.  Unless it's set, files would not be required (as it is now) and it could not be changed. I don't know that this is needed, since the toggle is restricted to superusers as it is, but thought it was worth the shout-out.

Reply:
@johannes-darms - is off by default good enough? Or is there a reason you'd want to completely shut off this API? (See discussion above).

Reply:
@qqmyers I commented on this based on the PR text, not the code. After writing my comment, I looked at the code and found a configuration option to disable the need for a data file, which wasn't clear from the text. So I deleted my comment.
We only store metadata elements for sensitive data, so we never have data files in the system. Whether this has at least one file requirement is enabled or disabled by default is not important to us. We just need an option to turn it off. IHMO requiring files is a reasonable default.

Reply:
Thanks. I think the default in the PR is still not to require files (so no change from prior version). In any case, sounds like no requirement for another flag.